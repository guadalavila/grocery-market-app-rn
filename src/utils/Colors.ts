export const Colors = {
    primary: '#68B984',
    primaryLight: '#CFFDE1',
    greenUltraLight: '#E9F9F2',
    green: '#2DA32A',
    secondary: '#3D5656',
    yellow: '#FED049',
    orange: '#FCB753',
    white: '#FFFFFF',
    black: '#000000',
    greyLight: '#C5C5C5',
    greyUltraLight: '#F7F8FA',
    background: '#FCFDFB',
    grey: '#9f9f9f',
    divider: '#EFEEF1',
};
