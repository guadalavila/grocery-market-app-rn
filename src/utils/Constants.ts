export const STORAGE_KEY = {
    FAVORITE: '@favorite',
};
