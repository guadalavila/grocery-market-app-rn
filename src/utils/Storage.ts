import AsyncStorage from '@react-native-async-storage/async-storage';

export const storeData = async (key: string, value: string) => {
    try {
        const jsonValue = JSON.stringify(value);
        await AsyncStorage.setItem(key, jsonValue);
    } catch (e) {}
};

export const getData = async (key: string) => {
    try {
        const data = await AsyncStorage.getItem(key);
        return data ? JSON.parse(data) : undefined;
    } catch (e) {
        // error reading value
    }
};
