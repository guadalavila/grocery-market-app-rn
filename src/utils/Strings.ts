export const Strings = {
    appName: 'Grocery Market',
    hello: 'Hola!',
    welcome: 'Bienvenido!',
    viewMore: 'Ver MÃ¡s',
    search: 'Buscar...',

    cart: {
        label: 'Carrito',
        empty: 'No tenes productos en el carrito',
        checkout: 'Terminar Compra',
        add: 'Agregar al carrito',
    },

    addedProduct: 'Producto Agregado!',

    fruits: {
        label: 'Frutas',
    },
    vegetables: {
        label: 'Verduras',
    },

    favorites: {
        label: 'Mis Favoritos',
        empty: 'No tenes productos como favoritos',
    },
};
