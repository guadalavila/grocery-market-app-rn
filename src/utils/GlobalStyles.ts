import { StyleSheet } from 'react-native';
import { Colors } from './Colors';

export const GlobalStyles = StyleSheet.create({
    container: {
        flex: 1,
    },
    shadowCard: {
        shadowColor: '#000',
        shadowOffset: {
            width: 0,
            height: 2,
        },
        shadowOpacity: 0.25,
        shadowRadius: 3.84,
        elevation: 4,
    },
    row: {
        flexDirection: 'row',
    },
    justBtw: {
        justifyContent: 'space-between',
    },
    main: {
        backgroundColor: Colors.background,
    },
});

export const flex = (x: number) => {
    return {
        flex: x,
    };
};

export const Sizes = {
    XXS: 2,
    XS: 4,
    S: 8,
    M: 16,
    L: 24,
    XL: 30,
    XXL: 40,
    XXXL: 55,
};

export const FontSize = {
    XS: 8,
    S: 10,
    M: 14,
    L: 18,
    XL: 22,
    XXL: 26,
    XXXL: 32,
};
