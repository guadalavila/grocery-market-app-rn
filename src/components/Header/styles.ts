import { StyleSheet } from 'react-native';
import { Colors } from '../../utils/Colors';
import { Sizes } from '../../utils/GlobalStyles';

export const styles = StyleSheet.create({
    container: {
        backgroundColor: 'transparent',
        paddingVertical: Sizes.M,
        flexDirection: 'row',
        paddingHorizontal: Sizes.S,
    },
    containerIcon: {
        width: '20%',
    },
    text: {
        fontWeight: 'bold',
        color: Colors.black,
        fontSize: 20,
        alignSelf: 'center',
        width: '60%',
        textAlign: 'center',
    },
});
