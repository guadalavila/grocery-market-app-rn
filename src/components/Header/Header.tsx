import { useNavigation } from '@react-navigation/native';
import React from 'react';
import { Text, View } from 'react-native';
import { TouchableOpacity } from 'react-native-gesture-handler';
import Icon from 'react-native-vector-icons/Ionicons';
import { Colors } from '../../utils/Colors';
import { styles } from './styles';

interface IHeaderProps {
    text: string;
    backButton: boolean;
}

const Header = ({ text, backButton }: IHeaderProps) => {
    const navigation = useNavigation();
    return (
        <View style={styles.container}>
            <View style={styles.containerIcon}>
                {backButton && (
                    <TouchableOpacity activeOpacity={0.7} onPress={() => navigation.goBack()}>
                        <Icon name='chevron-back-outline' size={32} color={Colors.grey} />
                    </TouchableOpacity>
                )}
            </View>
            <Text style={styles.text}>{text}</Text>
        </View>
    );
};

export default Header;
