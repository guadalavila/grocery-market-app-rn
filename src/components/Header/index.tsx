import React from 'react';
import { Text, View } from 'react-native';
import { styles } from './styles';

interface IHeaderProps {
    text: string;
}

const Header = ({ text }: IHeaderProps) => {
    return (
        <View style={styles.container}>
            <Text style={styles.text}>{text}</Text>
        </View>
    );
};

export default Header;
