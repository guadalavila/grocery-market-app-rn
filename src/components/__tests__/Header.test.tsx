import { fireEvent, render, screen } from '@testing-library/react-native';
import React from 'react';
import { Colors } from '../../utils/Colors';
import Header from '../Header/Header';

describe('Test Header', () => {
    test('should match snapshot', () => {
        const header = render(<Header text='Cart' backButton />);
        expect(header).toMatchSnapshot();
    });
    test('should render title', () => {
        render(<Header text='Cart' backButton />);
        expect(screen.getByText('Cart')).toBeTruthy();
    });

    test('should render icon', () => {
        render(<Header text='Cart' backButton />);
        screen.debug();
        expect(screen.getByTestId('icon')).toBeTruthy();
        expect(screen.getByTestId('icon').props.name).toBe('chevron-back-outline');
        expect(screen.getByTestId('icon').props.size).toBe(32);
        expect(screen.getByTestId('icon').props.color).toBe(Colors.black);
    });

    test('should render button', () => {
        render(<Header text='Cart' backButton />);
        const button = screen.getByTestId('back-button');
        expect(button).toBeTruthy();
    });

    test('should call fn when press button', () => {
        render(<Header text='Cart' backButton />);
        const button = screen.getByTestId('back-button');
        fireEvent.press(button);
    });
});
