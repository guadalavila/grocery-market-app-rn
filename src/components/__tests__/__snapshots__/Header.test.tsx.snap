// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header should match snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "flexDirection": "row",
      "paddingHorizontal": 8,
      "paddingVertical": 16,
    }
  }
>
  <View
    style={
      Object {
        "width": "20%",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="back-button"
    >
      <MockedIoniconsIcons
        color="#000000"
        name="chevron-back-outline"
        size={32}
        testID="icon"
      />
    </View>
  </View>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "#000000",
        "fontSize": 22,
        "fontWeight": "900",
        "textAlign": "center",
        "width": "60%",
      }
    }
  >
    Cart
  </Text>
</View>
`;
